### PR TITLE
Added micromasters config vars as pillar

### PR DESCRIPTION
**Pull request by shaidar**
#### What are the relevant tickets?
[Issue#1136](https://github.com/mitodl/salt-ops/issues/1136)

#### What's this PR do?
- Added keys/value to manage micromasters Heroku configs through the salt-proxy.
- Added config to top file
- Added config to salt-master

Note: Values referenced in vault would need to be populated prior to starting the salt-proxies

### CODE REVIEW

**Comment by shaidar** [comment]
@blarghmatey I remember you saying that this has another dependency, but can't recall what that is

**Comment by blarghmatey** [comment]
It's blocked on us provisioning the DB in the rc/prod apps environments or migrating the existing instance to those environments and then wiring it up with Vault.

**Comment by blarghmatey** [comment]
@shaidar this is now getting unblocked through the work that @feoh is doing. Can you sync up with him to get this PR updated with current settings and necessary fetches for Vault credentials, etc. so we can get it merged and managed via Salt the same way we are doing the other Heroku apps?

File: pillar/heroku/micromasters.sls [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      'ELASTICSEARCH_INDEX': 'micromasters',
+      'ELASTICSEARCH_SHARD_COUNT': 2,
+      'env_name': 'production',
+      'FEATURE_EXAMS_CARD_ENABLED': True,
+      'GA_TRACKING_ID': 'UA-5145472-10',
+      'GTM_CONTAINER_ID': 'GTM-NB27R8L',
+      'MAILGUN_FROM_EMAIL': 'no-reply@micromasters.odl.mit.edu',
+      'MAILGUN_URL': 'https://api.mailgun.net/v3/micromasters.odl.mit.edu',
+      'MICROMASTERS_BASE_URL': 'https://micromasters.mit.edu',
+      'MICROMASTERS_CORS_ORIGIN_WHITELIST': "['discussions.odl.mit.edu','odl.mit.edu']",
```
Comment (blarghmatey) [inline pillar/heroku/micromasters.sls:70]: Should the second entry be `open.mit.edu`?

File: pillar/heroku/micromasters.sls [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+proxy:
+  proxytype: heroku
+
+heroku:
+  app_name: {{ env_data.app_name }}
+  api_key: __vault__::secret-operations/heroku>data>api_key
+  config_vars:
+    ALLOWED_HOSTS: '{{ env_data.ALLOWED_HOSTS|tojson }}'
+    AWS_ACCESS_KEY_ID: __vault__:cache:aws-mitx/creds/read-write-delete-{{ business_unit }}-app-{{ env_data.env_name }}>data>access_key
```
Comment (blarghmatey) [inline pillar/heroku/micromasters.sls:102]: This should get updated to match the role defined in https://github.com/mitodl/ol-infrastructure/pull/1380

File: pillar/heroku/micromasters.sls [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    SECRET_KEY: __vault__:gen_if_missing:secret-{{ business_unit }}/django>data>secret_key    
+    SENTRY_AUTH_TOKEN: __vault__::secret-{{ business_unit }}/sentry>data>auth_token                 
+    SENTRY_DSN: __vault__::secret-{{ business_unit }}/sentry>data>dsn
+    SENTRY_ORG_NAME: 'mit-office-of-digital-learning'
+    SENTRY_PROJECT_NAME: 'micromasters'
+    STATUS_TOKEN:  __vault__:gen_if_missing:secret-{{ business_unit }}/django>data>status_token                     
+    UWSGI_PROCESS_COUNT: 4
+    UWSGI_SOCKET_TIMEOUT: 1                      
+    UWSGI_THREAD_COUNT: 50
+    {% if env_data.env_name == 'production' %}
```
Comment (blarghmatey) [inline pillar/heroku/micromasters.sls:173]: Should this get updated to be conditional on RC and production, now that we have the additional stacks in Pulumi?

File: pillar/heroku/micromasters.sls [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    {% set pg_creds = salt.vault.cached_read('postgres-micromasters/creds/app', cache_prefix='heroku-micromasters') %}
+    ADWORDS_CONVERSION_ID: 935224753
+    CLIENT_ELASTICSEARCH_URL: '/api/v0/search/'
+    DATABASE_URL: postgres://{{ pg_creds.data.username }}:{{ pg_creds.data.password }}@{{ rds_endpoint }}/micromasters
+    FEATURE_PEARSON_EXAMS_SYNC: True
+    {% endif %}
+    {% if env_data.env_name != 'ci' %}
+    CLOUDFRONT_DIST: {{ env_data.CLOUDFRONT_DIST }}
+    ENABLE_STUNNEL_AMAZON_RDS_FIX: True
+    EXAMS_AUDIT_AWS_ACCESS_KEY_ID: __vault__:cache:aws-mitx/creds/read-write-{{ business_unit }}-app-{{ env_data.env_name }}>data>access_key
```
Comment (blarghmatey) [inline pillar/heroku/micromasters.sls:183]: Is this encapsulated in the Pulumi managed role now?

**Review by blarghmatey — COMMENTED** [review]
Just a few questions/notes, otherwise looks great :)

File: pillar/heroku/micromasters.sls (hunk 1 above)
Comment (shaidar) [reply pillar/heroku/micromasters.sls:70]: Currently that's how it's set in the Heroku app, which I don't think makes a lot of sense. Will actually just change it to `open.mit.edu`

File: pillar/heroku/micromasters.sls (hunk 4 above)
Comment (shaidar) [reply pillar/heroku/micromasters.sls:183]: Will store the current creds statically in vault as this is no longer in use and only a matter of tech debt - https://github.com/mitodl/micromasters/issues/5314

**Pull request merged**